### DISCUSSION TITLE
onnx inference speed

### DISCUSSION THREAD

**Discussion by xIaott-s** [speechbrain/speechbrain#1857] [Q&A]
Hi, I export the ecapa-tdnn model to onnx and use onnxruntime to inference. But the inference speed is slower than pytorch. Does anybody konw why?

**Comment by aheba** [comment]
@sty1992928 , as far as I know,
Generally, using ONNX + TensorRT improve by high range the performance,

are you using this project ?
https://github.com/NVIDIA-AI-IOT/torch2trt

**Comment by vadimkantorov** [comment]
Onnxruntime has some unsolved perf problems with conv1d: https://github.com/microsoft/onnxruntime/issues/7212 https://github.com/microsoft/onnxruntime/issues/8513

We found only version 1.8.0 is somewhat performant (usually these issues are about bad configuration of cudnn / cudnn benchmarking workspace size). Please also report to onnxruntime to bump these issues in priority

**Comment by xIaott-s** [comment]
> are you using this project ?

I use cpu only，is this as expected？

**Comment by xIaott-s** [comment]
> Onnxruntime has some unsolved perf problems with conv1d: [microsoft/onnxruntime#7212](https://github.com/microsoft/onnxruntime/issues/7212) [microsoft/onnxruntime#8513](https://github.com/microsoft/onnxruntime/issues/8513)
> 
> We found only version 1.8.0 is somewhat performant (usually these issues are about bad configuration of cudnn / cudnn benchmarking workspace size). Please also report to onnxruntime to bump these issues in priority

I use cpu only, does cpu mode has the same perf problems?

**Comment by vadimkantorov** [comment]
Those issues are unrelated to CPU, so likely not...

**Comment by xIaott-s** [comment]
> Those issues are unrelated to CPU, so likely not...

Thank you  anyway！The Int8 quant works，so my problem is temporarily gone.

**Comment by xle97** [comment]
> Hi, I export the ecapa-tdnn model to onnx and use onnxruntime to inference. But the inference speed is slower than pytorch. Does anybody konw why?

hi，May I ask how you exported onnx？

**Comment by aheba** [comment]
Hi you can use torch.onnx.export to do the onnx export of torch module;
Basically you will need for the forward + a tracing example..

**Comment by marziye-A** [comment]
> Hi, I export the ecapa-tdnn model to onnx and use onnxruntime to inference. But the inference speed is slower than pytorch. Does anybody konw why?

dear @xIaott-s, can you please share your code on how you have converted your model to onnx and quantized it?
